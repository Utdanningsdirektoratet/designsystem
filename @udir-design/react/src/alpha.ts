/*
 * This is the entry point for the /alpha sub-module.
 * It exports the alpha-stage components, plus everything in beta (& stable).
 */
export * from './beta';
export * from './components/alpha';
export * from './utilities/hooks/alpha';
