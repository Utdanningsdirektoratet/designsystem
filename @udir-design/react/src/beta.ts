/*
 * This is the entry point for the /beta sub-module.
 * Only beta components should be exported here.
 */

export * from './components/beta';
