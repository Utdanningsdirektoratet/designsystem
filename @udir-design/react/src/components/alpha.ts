/*
 * This file exports the components that are in the alpha stage of development
 */

export * from './button/Button';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './details/Details';
export * from './dialog/Dialog';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './errorSummary/ErrorSummary';
export * from './field/Field';
export * from './fieldset/Fieldset';
export * from './input/Input';
export * from './link/Link';
export * from './list/List';
export * from './loaders';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skipLink/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
export * from './textarea/Textarea';
export * from './textfield/Textfield';
export * from './toggleGroup/ToggleGroup';
export * from './tooltip/Tooltip';
export * from './typography';
