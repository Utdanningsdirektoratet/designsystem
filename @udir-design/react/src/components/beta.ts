/*
 * This file exports the components that are in the beta stage of development
 */

export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './button/Button';
export * from './card/Card';
export * from './chip/Chip';
export * from './checkbox/Checkbox';
export * from './details/Details';
export * from './dialog/Dialog';
export * from './divider/Divider';
export * from './errorSummary/ErrorSummary';
export * from './field/Field';
export * from './fieldset/Fieldset';
export * from './input/Input';
export * from './link/Link';
export * from './list/List';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skeleton/Skeleton';
export * from './skipLink/SkipLink';
export * from './spinner/Spinner';
export * from './tag/Tag';
export * from './textarea/Textarea';
export * from './textfield/Textfield';
export * from './tooltip/Tooltip';
