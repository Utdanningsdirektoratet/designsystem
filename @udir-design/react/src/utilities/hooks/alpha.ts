/*
 * This file exports the hooks that are in the alpha stage of development
 */

export {
  type UsePaginationProps,
  usePagination,
} from './usePagination/usePagination';
export {
  useTableOfContents,
  type useTableOfContentsProps,
} from './useTableOfContents/useTableOfContents';
