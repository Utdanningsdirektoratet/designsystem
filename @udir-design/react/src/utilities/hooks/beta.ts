/*
 * This file exports the hooks that are in the beta stage of development
 */

export {
  type UseCheckboxGroupProps,
  useCheckboxGroup,
} from './useCheckboxGroup/useCheckboxGroup';
export {
  type UseRadioGroupProps,
  useRadioGroup,
} from './useRadioGroup/useRadioGroup';
