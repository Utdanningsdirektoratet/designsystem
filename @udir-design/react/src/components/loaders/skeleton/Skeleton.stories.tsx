import type { Meta, StoryObj } from '@storybook/react-vite';
import { Button, Skeleton } from '../../beta';
import { Heading, Paragraph } from '../../alpha';

const meta: Meta<typeof Skeleton> = {
  component: Skeleton,
  tags: ['alpha'],
  parameters: {
    a11y: {
      config: {
        // Disable a11y empty heading rule as we intentionally set aria-hidden="true" on the Skeleton component inside Headings
        rules: [{ id: 'empty-heading', selector: ':has(.ds-skeleton)' }],
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Skeleton>;

export const Preview: Story = {
  args: {
    width: 200,
    height: 100,
  },
};

export const Components: Story = {
  args: {
    height: 50,
  },
  render: (args) => {
    return (
      <div
        style={{
          display: 'flex',
          alignItems: 'center',
          gap: '20px',
        }}
      >
        <Skeleton variant="circle" width="50px" {...args} />
        <Skeleton variant="rectangle" width="100px" {...args} />
        <Paragraph>
          <Skeleton variant="text" width="10" {...args} />
        </Paragraph>
      </div>
    );
  },
};

export const UsageExample: Story = {
  render: (args) => {
    return (
      <div
        style={{
          maxWidth: 400,
        }}
      >
        <Skeleton height="150px" {...args} />
        <div
          style={{
            display: 'flex',
            gap: '10px',
            alignItems: 'center',
            padding: '5px 0 5px 0',
          }}
        >
          <Skeleton variant="circle" width="30px" height="30px" {...args} />
          <Heading>
            <Skeleton variant="text" {...args}>
              En medium tittel
            </Skeleton>
          </Heading>
        </div>
        <Skeleton variant="text" width="140" />
      </div>
    );
  },
};

export const Children: Story = {
  args: {
    variant: 'rectangle',
    children: [
      <Paragraph>
        Her er en tekst som blir sendt inn som barn av en Skeleton.
      </Paragraph>,
      <Paragraph>
        Se hvordan Skeleton da dekker den samlede bredden og høyden til barna.
      </Paragraph>,
      <Button>Knapp</Button>,
    ],
  },
};

export const Text: Story = {
  args: {
    variant: 'text',
  },
  render: (args) => (
    <div style={{ display: 'flex', gap: '20px', maxWidth: 300 }}>
      <div style={{ flex: '1 1 200px' }}>
        <Heading>En tittel</Heading>
        <Paragraph data-size="sm">
          Her er en paragraf som går over flere linjer
        </Paragraph>
      </div>
      <div style={{ flex: '1 1 200px' }}>
        <Heading>
          <Skeleton {...args}>En tittel</Skeleton>
        </Heading>
        <Paragraph data-size="sm">
          <Skeleton width={44} {...args} />
        </Paragraph>
      </div>
    </div>
  ),
};
