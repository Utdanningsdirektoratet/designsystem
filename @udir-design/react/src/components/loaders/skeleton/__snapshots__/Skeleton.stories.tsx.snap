// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Preview 1`] = `
<span
  aria-hidden="true"
  data-variant="rectangle"
>
</span>
`;
